distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/model/WorkItem.java

```diff
@@ -1,17 +1,19 @@
 package ru.fix.distributed.job.manager.model;
 
+import kotlin.text.Regex;
 import ru.fix.distributed.job.manager.JobId;
 
 import java.util.Objects;
 
 public class WorkItem {
-
-    private String id;
-    private JobId jobId;
+    private static final Regex PATTERN = new Regex("[a-zA-Z0-9_-[.]]+");
+    private final String id;
+    private final JobId jobId;
 
     public WorkItem(String id, JobId jobId) {
         Objects.requireNonNull(id);
         Objects.requireNonNull(jobId);
+        assert PATTERN.matches(id);
```

Comment:
why you used _assert_ instead of _require_? in _require_ we can set specific message in IllegalArgumentException

Reply:
fixed